cpmpy/solvers/TEMPLATE.py

```diff
@@ -257,3 +257,51 @@ def _post_constraint(self, cpm_con):
 
     # Other functions from SolverInterface that you can overwrite:
     # solveAll, solution_hint, get_core
+
+    def solveAll(self, display=None, time_limit=None, solution_limit=None, **kwargs):
+        """
+            A shorthand to (efficiently) compute all (optimal) solutions, map them to CPMpy and optionally display the solutions.
+
+            If the problem is an optimization problem, returns only optimal solutions.
+
+           Arguments:
+                - display: either a list of CPMpy expressions, OR a callback function, called with the variables after value-mapping
+                        default/None: nothing displayed
+                - time_limit: stop after this many seconds (default: None)
+                - solution_limit: stop after this many solutions (default: None)
+                - any other keyword argument
+
+            Returns: number of solutions found
+        """
+
+        # check if objective function?
+        if self.TEMPLATE_solver.HasObjective():
+            raise Exception("TEMPLATE does not support finding all optimal solutions")
+
+        # A. Example code if solver supports callbacks
+        if is_any_list(display):
+            callback = lambda : print([var.value() for var in display])
+        else:
+            callback = display
+
+        self.solve(time_limit, callback=callback, enumerate_all_solutions=True, **kwargs)
```

Comment:
If this is the template implementation for `solveAll`, I don't see an advantage to calling `solve` instead of manually blocking the found solutions. See [comment](https://github.com/CPMpy/cpmpy/pull/176#issuecomment-1368747103) for ideas on `solveAll`.

Reply:
```Python
        if time_limit is not None:
            raise NotImplementedError("TEMPLATE: TODO, implement time_limit")
```

I would use a warning for this and have a default implementation in `solverInterface` with `time_limit`.

Reply:
`216:        raise NotImplementedError("TEMPLATE: Not a know supported numexpr {}".format(cpm_expr))`

KNOW -> KNOWN

Reply:
**Line 234:**
I would at least write the transformations that can be done in the order that they should be applied to something similar to `ortools.py`:
```

cpm_cons = flatten_constraint(cpm_con)
cpm_cons = reify_rewrite(cpm_cons)
cpm_cons = only_numexpr_equality(cpm_cons, supported={"sum", "wsum","sub"})
cpm_cons = only_bv_implies(cpm_cons) # everything that can create
```

Reply:
We should do the changes you proposed in TEMPLATE.py directly in master.
Regarding calling `solve` in `solveAll`, this is actually the way both OR-tools and Gurobi handle the solve all call